lib/hal/hal_ssd1608.ex

```diff
@@ -100,17 +94,19 @@ defmodule Inky.HAL.PhatSSD1608 do
     end
   end
 
-  @spec do_update(State.t(), Inky.Display.t(), atom(), binary(), binary()) :: :ok
-  defp do_update(state, _display, border, black_bits, accent_bits) do
+  @spec do_update(State.t(), atom(), binary(), binary()) :: :ok
+  defp do_update(state, border, black_bits, accent_bits) do
+    %{width: width, height: height, luts: luts} = state.display
+
     state
-    |> write_command(@cmd_set_driver_output, [@rows - 1, (@rows - 1) >>> 8, 0x00])
+    |> write_command(@cmd_set_driver_output, [width - 1, (width - 1) >>> 8, 0x00])
     |> write_command(@cmd_set_dummy_line_period, [0x1B])
     |> write_command(@cmd_set_gate_line_width, [0x0B])
     |> write_command(@cmd_set_data_entry_mode, [0x03])
-    |> write_command(@cmd_set_ram_x_position, [0x00, div(@cols, 8) - 1])
-    |> write_command(@cmd_set_ram_y_position, [0x00, 0x00, @rows - 1, (@rows - 1) >>> 8])
+    |> write_command(@cmd_set_ram_x_position, [0x00, div(height, 8) - 1])
+    |> write_command(@cmd_set_ram_y_position, [0x00, 0x00, width - 1, (width - 1) >>> 8])
     |> write_command(@cmd_write_vcom, [0x70])
-    |> write_command(@cmd_write_lut, @lut_data)
+    |> write_command(@cmd_write_lut, luts)
```

Comment:
What do these changes contribute? The point of the refactor was to not spend runtime unpacking constants that are known at compile time.

Reply:
That makes sense. I'll keep lut data in the HAL module.

Reply:
Another question regarding the LUTS, are they the same for both displays?

Reply:
The pHat SSD1608 uses the same lut data regardless of the third color. 
https://github.com/pimoroni/inky/blob/fc17026df35447c1147e9bfa38988e89e75c80e6/library/inky/inky_ssd1608.py#L106-L121